floyd/client/files.py

```diff
@@ -119,45 +122,104 @@ def sizeof_fmt(num, suffix='B'):
     return "%.1f%s%s" % (num, 'Yi', suffix)
 
 
-def warn_purge_exit(info_msg, filename, exit_msg):
+class DataCompressor(object):
     """
-    Warn the user that's something went wrong,
-    remove the tarball and provide an exit message.
+    Local Data Compression with progress bar.
     """
-    floyd_logger.info(info_msg)
-    rmtree(os.path.dirname(filename))
-    sys.exit(exit_msg)
-
+    def __init__(self,
+                 source_dir,
+                 filename="/tmp/contents.tar.gz"):
+        # Data directory to compress
+        self.source_dir = source_dir
+        # Archive (Tar file) name
+        self.filename = filename
+
+        # Prgress Bar for tracking data compression
+        self.__compression_bar = None
+
+        # Number of files to compress
+        self.__files_to_compress = 0
+        self.__get_nfiles_to_compress()
+
+    def __get_nfiles_to_compress(self):
+        """
+        Return the number of files to compress
+        """
+        floyd_logger.info("Get number of files to compress... (it could take some seconds)")
+        paths = [self.source_dir]
+        try:
+            # Traverse each subdirs of source_dir and count files/dirs
+            while paths:
+                path = paths.pop()
+                for item in scandir(path):
+                    if item.is_dir():
+                        paths.append(item.path)
+                        self.__files_to_compress += 1
+                    elif item.is_file():
+                        self.__files_to_compress += 1
+        except OSError as e:
+            # OSError: [Errno 13] Permission denied
+            if e.errno == errno.EACCES:
+                self.source_dir = os.getcwd() if self.source_dir == '.' else self.source_dir  # Expand cwd
+                sys.exit(("Permission denied. Make sure to have read permission "
+                          "for all the files and directories in the path: %s")
+                         % (self.source_dir))
+        floyd_logger.info("Compressing {} files".format(self.__files_to_compress))
```

Comment:
better practice for logging is: `floyd_logger.info("Compressing % files", self.__files_to_compress)`